Add default facts for puppet-server provisioner

seanmalloy opened the pull request:
This is a new feature.

Adding default facts to the puppet-server provisioner. New default facts are named packer_build_name and packer_builder_type. This feature was implemented for the puppet-masterless provisioner in #1878.

I was able to build the packer binary with the code changes that I made. I have not tested the changes yet. I am hoping to test the changes in the next day or two. I will also write some unit tests for this change.

Let me know if you see anything glaringly wrong with the proposed changes.

rickard-von-essen commented:
Thanks! 

Remove WIP and comment when you have tested this and added a unit test.

seanmalloy commented:
I successfully tested the change manually using "packer build".

I am hoping to write the unit tests in the next week. I will let you now when I have the unit tests written.

seanmalloy commented:
Everything should be ready now.

I created the unit tests. I tested it using "packer build" against my puppet master. I rebased my branch. I removed WIP.

Let me know if you find any problems.

rickard-von-essen commented:
LGTM!
